n_queen_problem/NQueenProblem.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Following code is the implementation of the N-queen problem.
+ * The solution is to place the queens on board such that no queen is under attack by another queen.
+ * There can be a number of possible solutions for a given length of board.
+ * This implementation prints only one valid solution, it can be extended to print all possible valid solutions.
+ */
+
+import java.util.Arrays;
+
+public class NQueenProblem {
+
+    private static final boolean isQueenSafe(int[][] board, int row, int column) {
```

Comment:
Why have you added `final` to `private` methods if they can't be overridden? I think this practice causes unnecessary confusion and is redundant.